docs/backend-system/building-backends/08-migrating.md

```diff
@@ -7,3 +7,238 @@ description: How to migrate existing backends to the new backend system
+ backend.start();
+```
+
+The `todo` plugin used above is just an example and you may not have it enabled
+in your own backend. Feel free to change it to some other plugin that you
+actually have in your `plugins` folder, for example
+`backend.add(legacyPlugin('catalog', import('./plugins/catalog')))`.
+
+The `legacyPlugin` helper makes it easy to bridge the gap between the old-style
+plugin files and the new backend system. It ensures that the dependencies that
+you used to have to declare by hand in your env are gathered behind the scenes,
+then passes them into the relevant `createPlugin` export function, and makes
+sure that the route handler it returns is passed into the HTTP router with the
+given prefix.
+
+# Handling Custom Environments
+
+In the simple case, what we did above is sufficient, TypeScript is happy, and
+the backend runs with the new feature. If they do, feel free to skip this entire
+section.
+
+Sometimes though, type errors can be reported on the newly added line, saying
+that parts of the `PluginEnvironment` type do not match. This happens when the
+environment was changed from the defaults, perhaps with your own custom
+additions. If this is the case in your installation, you still aren't out of
+luck - you can build a customized `legacyPlugin` function.
+
+```diff
+ import { createBackend } from '@backstage/backend-defaults';
+-import { legacyPlugin } from '@backstage/backend-common';
++import { makeLegacyPlugin, loggerToWinstonLogger } from '@backstage/backend-common';
++import { coreServices } from '@backstage/backend-plugin-api';
+
++const legacyPlugin = makeLegacyPlugin(
++  {
++    cache: coreServices.cache,
++    config: coreServices.config,
++    database: coreServices.database,
++    discovery: coreServices.discovery,
++    logger: coreServices.logger,
++    permissions: coreServices.permissions,
++    scheduler: coreServices.scheduler,
++    tokenManager: coreServices.tokenManager,
++    reader: coreServices.urlReader,
++    identity: coreServices.identity,
++    // ... and your own additions
++  },
++  {
++    logger: log => loggerToWinstonLogger(log),
++  },
++);
+
+ const backend = createBackend();
+ backend.add(legacyPlugin('todo', import('./plugins/todo')));
+ backend.start();
+```
+
+The first argument to `makeLegacyPlugin` is the mapping from environment keys to
+references to actual [backend system services](../architecture/03-services.md).
+The second argument allows you to "tweak" the types of those services to
+something more fitting to your env. For example, you'll see that the logger
+service API type was changed from the raw Winston logger of old, to a different,
+custom API, so we use a helper function to transform that particular one.
+
+To make additions as mentioned above to the environment, you will start to get
+into the weeds of how the backend system wiring works. You'll need to have a
+service reference, and a service factory that performs the actual creation of
```

Comment:
```suggestion
service reference and a service factory that performs the actual creation of
```

Reply:
isn't this proper use of [oxford comma](https://www.grammarly.com/blog/what-is-the-oxford-comma-and-why-do-people-care-so-much-about-it/)

Reply:
Nah oxford comma is for when you're listing at least 3 things right? This is connecting two clauses together and imo adding an unnecessary pause

Reply:
I think that's a swedish rule. But sure I can remove anyway